portfolio/src/main/webapp/script.js

```diff
@@ -63,21 +63,51 @@ function previousExperience() {
   selectedExperience--;
 }
 
-function getServerComments() {
-  fetch('/data').then(response => response.json()).then((commentSection) => {
+function getNumCommentsSelected(selected) {
+  var selectedValue = selected.value;
+  getServerComments(selectedValue);
+}
+
+function getServerComments(numCommentsSelected) {
+  fetch('/data?number-comments='+numCommentsSelected).then(response => response.json()).then((commentSection) => {
     populateComments(commentSection);
   });
 }
 
 function populateComments(commentSection) {
   const individualComments = document.getElementById('comments-container');
+  individualComments.innerHTML = '';
   commentSection.forEach((element) => {
-      individualComments.appendChild(createComment(element));
+    individualComments.appendChild(createComment(element));
   });
+  if (sessionStorage.getItem('selectedValue') === null){
+    document.getElementById('number-comments').options[1].selected = true;
+  } else {
+    document.getElementById('number-comments').options[sessionStorage.getItem('selectedValue')].selected = true;
+  }
 }
 
 function createComment(text) {
   const liComment = document.createElement('li');
   liComment.innerText = text;
   return liComment;
+}
+
+function deleteComments() {
+  fetch('/delete-data', {method: 'POST'}).then(getServerComments(0));
+}
+
+function storeSelectedValue(selected) {
+  sessionStorage.setItem('selectedValue', selected.value);
+  if (sessionStorage.getItem('selectedValue')) {
+      document.getElementById('number-comments').options[sessionStorage.getItem('selectedValue')].selected = true;
+  }
+}
```

Comment:
The only difference between `sessionStorage` and using a variable here would be that `sessionStorage` persists when you change pages on your site and then come back to this page, whereas the variable would get reset. Was this a deliberate design choice on your part or did you end up using (`localStorage` earlier, now `sessionStorage`) that for some other reason, eg. the previously embedded JS code in the HTML wouldn't work without it?

Reply:
This is not so much a request for changes, it's more of an FYI and to ask if you knew this/intended for this beforehand.

Reply:
I used storage because when the page reloaded (after a new comment was added for example), I wanted the selected drop down to stay. Not so much for switching pages on the site, but more so for refreshing this particular page. Are there drawbacks to this / do you think this might be unnecessary? (you don't have to answer, but I'd be curious and happy to know your input)

Reply:
If you can avoid reloading the page, that would be best. If it seems like more than a brief effort, then you can skip it for the time being. In general, it's good to avoid full page reloads where possible. Modern websites tend to fetch data from the backend piece by piece and iteratively update the UI in small chunks. This means less data is transferred between the backend and the frontend and has the user spending less time waiting for things to load/update.

Reply:
Okay, that makes sense and is good to know, thank you. I might wait on making this change so that I can make sure I deploy last week's work and can start this week's tutorial, but I will remember this.